vault/resource_database_secret_backend_connection.go

```diff
@@ -831,6 +836,13 @@ func getMSSQLConnectionDetailsFromResponse(d *schema.ResourceData, prefix string
 		}
 		result[0]["contained_db"] = containedDB
 	}
+	if v, ok := details["username"]; ok {
```

Comment:
To reduce some duplication, perhaps replace the `getConnectionDetailsFromResponse()` call with `getUserPassConnectionDetailsFromResponse()`

Reply:
Nice catch! Fixed in 98e6985